### ISSUE TITLE
[Bug] Crash on using TM

### ISSUE THREAD

**Issue by ByThePowerOfScience** [pagefaultgames/pokerogue#4107]
### Describe the bug

I used the TM for Payback on my Chespin in slot 2 (1-indexed), and the game went to a black screen.  Checking the console revealed the following stacktrace:
```
TypeError: t.usedTMs is undefined
    start https://pokerogue.net/assets/index-xHZ0xkVd.js:15689
    onActionInput https://pokerogue.net/assets/index-xHZ0xkVd.js:15406
    processInput https://pokerogue.net/assets/index-xHZ0xkVd.js:15412
    processInput https://pokerogue.net/assets/index-xHZ0xkVd.js:15722
    buttonAb https://pokerogue.net/assets/index-xHZ0xkVd.js:16250
    5 https://pokerogue.net/assets/index-xHZ0xkVd.js:16250
    listenInputs https://pokerogue.net/assets/index-xHZ0xkVd.js:16250
    emit https://pokerogue.net/assets/index-xHZ0xkVd.js:1
    gamepadButtonDown https://pokerogue.net/assets/index-xHZ0xkVd.js:16250
    emit https://pokerogue.net/assets/index-xHZ0xkVd.js:1
    update https://pokerogue.net/assets/index-xHZ0xkVd.js:3239
    update https://pokerogue.net/assets/index-xHZ0xkVd.js:3243
    refreshPads https://pokerogue.net/assets/index-xHZ0xkVd.js:3247
    update https://pokerogue.net/assets/index-xHZ0xkVd.js:3247
    emit https://pokerogue.net/assets/index-xHZ0xkVd.js:1
    updatePoll https://pokerogue.net/assets/index-xHZ0xkVd.js:3031
    preRender https://pokerogue.net/assets/index-xHZ0xkVd.js:3027
    emit https://pokerogue.net/assets/index-xHZ0xkVd.js:1
    step https://pokerogue.net/assets/index-xHZ0xkVd.js:466
    step https://pokerogue.net/assets/index-xHZ0xkVd.js:470
    d https://pokerogue.net/assets/index-xHZ0xkVd.js:1036
[index-xHZ0xkVd.js:16376:31258](https://pokerogue.net/assets/index-xHZ0xkVd.js)
    onerror https://pokerogue.net/assets/index-xHZ0xkVd.js:16376
```

### Session export file

Changed to `txt` so GitHub would let me upload it: [sessionData_ABadHaiku.prsv.txt](https://github.com/user-attachments/files/16922456/sessionData_ABadHaiku.prsv.txt)

### User data export file

Likewise here: [data_ABadHaiku.prsv.txt](https://github.com/user-attachments/files/16922459/data_ABadHaiku.prsv.txt)

### Expected behavior

The TM to apply and the game to not crash.

Don't you just love JS? \:P

### Screenshots / Videos

![image](https://github.com/user-attachments/assets/26f6112c-a403-4155-886d-f16323ea7b86)

### What OS did you observe the bug on?

PC/Windows

### Which browser do you use?

Firefox

### Additional context

Does not seem to be consistent.  After restarting the game, the TM worked fine.

Probably just needs a safeguard there; I'd make the PR myself if I knew where in the source it was.

**Comment by Snailman11** [comment]
Link to Discord Report (Learning a new move hangs.)
https://discord.com/channels/1125469663833370665/1282399280228864092

**Comment by MokaStitcher** [comment]
Fixed with #4132
